users/views.py

```diff
@@ -124,4 +125,18 @@ def servertime(self, *args, **kwargs):
         """ calendar endpoint.
             globally used datetime.
         """
-        return Response(self.get_server_time(), status=200)
\ No newline at end of file
+        return Response(self.get_server_time(), status=200)
+
+
+class TimeClock(Query, Slack, ViewSet):
+    """ for Clock-in and Clock-out 
+    """
+    
+    def post(self, *args, **kwargs):
+        # used by slack api
+        serializer = TimeLogSerializer(data=self.request.data)
+        serializer.is_valid(raise_exception=True)
+        # returns if user clocked-in or clocked-out
+        msg = serializer.save()
```

Comment:
return the timelog data and not just a string message. or maybe return nothing at all. http204